drivers/video/video_display_legacy.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2019, Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * video_display_legacy is a shim layer allowing to expose a display compatible
```

Comment:
Wondering if it would not be a better idea to add a generic video_display_driver to `drivers/display` that forwards the calls to an associated video display device (a driver in `drivers/video`) in this way no extra code needs to be added to the driver in `drivers/video`.

Further I dont think it is a good idea to call it `legacy` as far as I can see the `video` API assumes it operates on full frame buffers where as the read/write function in the `display` API are designed to work with partial buffer as a lot of devices will not have enough RAM to operate with full frame buffers.

Reply:
yes will change this misleading name

Reply:
@vanwinkeljan , could you please review again

Reply:
@loicpoulain had a quick look but should the new implementation not create a display device that has a pointer to the video device? All calls made to the de display device can then be forwarded to the video device.

Reply:
well, for me we just have one device implementing two APIs. Adding a new dummy display device is possible but I'm not sure what it would bring... except maybe removing some display specific code from the video driver.

Reply:
To me it looks like now a third API has been introduced:
1) display API
2) video API
3) display video API

I thought the intention of the shim layer was to reuse existing display code.

Or is just a DEVICE_AND_API_INIT missing in `drivers/video/video_mcux_elcdif.c`  to register the shim layer?

Reply:
> To me it looks like now a third API has been introduced:
> 
>     1. display API
> 
>     2. video API
> 
>     3. display video API
> 
> 
> I thought the intention of the shim layer was to reuse existing display code.
> 
> Or is just a DEVICE_AND_API_INIT missing in `drivers/video/video_mcux_elcdif.c` to register the shim layer?

No still 2 APIs, It's just that video_mcux_elcdif has an option/config to also support 'display API' (via shim layer). So at the end, there is one device implementing both video and display APIs.

Reply:
Problem that I currently see is that the display API assumes that a device driver implements the display API. So if you pass a `dev` to any of display API functions it casts the `driver_api` member to `struct display_driver_api` which is not valid for the proposed `video_mcux_elcdif` driver as the `driver_api` member for `video_mcux_elcdif` driver is of the type `struct video_driver_api`


https://github.com/zephyrproject-rtos/zephyr/blob/2d9032e5d09e12ed8efc1132b1666583c9689895/include/drivers/display.h#L239-L248

Reply:
the **dev->driver_api** can be either cast as **video_driver_api** or **display_driver_api** because video_driver_api inherits display_driver_api (cf include/drivers/video.h).

https://github.com/zephyrproject-rtos/zephyr/blob/064930a565065c53f256a99bfbd88ce766e2c442/include/drivers/video.h#L233

Reply:
ha that was the line I missed :)

Thx for pointing out and sorry for all the miscommunication